feat(studio): Add expiration date and last used to personal access tokens

kamilogorek opened the pull request:
FE part of https://linear.app/supabase/issue/API-605/add-optional-expiration-date-to-newly-created-pats

Requires https://github.com/supabase/infrastructure/pull/24939 to be deployed first.

Reworked table to use not deprecated `Table` component along the way.

It would be nice if we could use custom calendar dropdown later, but this was beyond me 😢 

<img width="480" height="362" alt="image" src="https://github.com/user-attachments/assets/b6643edf-51a7-40ce-8ec6-feacae9cfea4" />

<img width="1033" height="716" alt="image" src="https://github.com/user-attachments/assets/7ed8c83e-05ea-4434-b683-993b19128c18" />

kamilogorek committed:
feat(studio): Add expiration date and last used to personal access tokens

joshenlim committed:
Merge branch 'master' into expiring-pats

joshenlim committed:
Nit polish

joshenlim reviewed (approved):
Tested on staging preview (payload sent when creating PAT includes expires at if an expiry is selected, and doesnt have that property if "never" is selected)

LGTM! Approving to unblock but will mark as draft for now since its blocked on API changes

kamilogorek committed:
ref: Mark expired tokens as such

kamilogorek committed:
Merge branch 'master' into expiring-pats

kemaldotearth committed:
feat: updated ui with expiry date custom date

As we had some merge conflicts this tidies that up as well as adds custom date picker for the expiry date.

kemaldotearth committed:
chore: run prettier

joshenlim committed:
Merge branch 'master' into expiring-pats

joshenlim committed:
Clean up

joshenlim committed:
Nit

joshenlim reviewed (approved):
Tested on staging preview and locally! 🙂
